File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
    // 배포 진도 순회?
      let days = progresses.map((progresses, idx) => Math.ceil((100 - progresses) / speeds[idx]));
    
    // 배포 가능일에 따라 n개의 기능이 배포되는지 산출
    let maxDay = 0;
    let cnt = 1;
    let ans = []
    for(let val of days) {
        if(maxDay === 0) maxDay = val;
        else if(maxDay >= val) cnt++;
        else {
            ans.push(cnt);
            maxDay = val;
            cnt = 1;
        }
    }
    ans.push(cnt)
    return ans;
}

File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    answer = []
    while progresses:
        cnt = 0
        for idx in range(len(progresses)):
            progresses[idx] += speeds[idx]
        while progresses and progresses[0] >= 100:
            progresses.pop(0)
            speeds.pop(0)
            cnt += 1
        if cnt > 0: answer.append(cnt)
    return answer